src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -1685,6 +1693,106 @@ func (b PCIeRootPortDevice) Valid() bool {
 	return true
 }
 
+// PCIeSwitchUpstreamPortDevice is the port connecting to the root port
+type PCIeSwitchUpstreamPortDevice struct {
+	ID  string // format: sup{n}, n>=0
+	Bus string // default is rp0
+}
+
+// QemuParams returns the qemu parameters built out of the PCIeSwitchUpstreamPortDevice.
+func (b PCIeSwitchUpstreamPortDevice) QemuParams(config *Config) []string {
```

Comment:
Just curious why the name `b`? That evokes `bus`. Is that what you wanted? I would try to keep the variable name and type name consistent, i.e. use 'device' / 'd' in both.